### DISCUSSION TITLE
Rendered message acting as if it was a message template?

### DISCUSSION THREAD

**Discussion by chuseman** [datalust/seq-tickets#1984] [Help] [answered]
We've got some clients logging only a rendered message, not a message template. Here's an example of what they're sending:

```json
{"@t":"2023-10-11T14:30:00.0000000Z","@m":"This is a test message {foo: 'bar'}","@i":"01234567"}
```

When I view these logs, the `{foo: 'bar'}` bit shows as just `{foo}` but is rendered in red (maybe indicating a malformed message template?):

![image](https://github.com/datalust/seq-tickets/assets/1007851/509e93ac-55c7-4987-85d8-62f31d6fb8a8)

So there's no way to see the original `{foo: 'bar'}` bit.

If I click **Export -> Copy raw JSON**, I get this:

```json
{"@t":"2023-10-11T14:30:00.0000000Z","@mt":"This is a test message {foo: 'bar'}","@m":"This is a test message {foo: 'bar'}","@i":"01234567"}
```

It looks like the message template `@mt` property is being copied from the rendered message `@m` property? That seems odd - the client didn't send a message template at all.

Is there a way to make these visible in the log viewer? I can't easily change the format of these legacy log messages or transform them to actual message templates.

**Comment by nblumhardt** [comment, marked as answer]
Thanks for reporting this, Chris.

This is a known bug of considerable vintage; it's been left this way for some time to avoid breakage, but coming back to it now I think we'd be better off just making the change, here. I'll raise a ticket referencing this thread. The fix will likely need to wait until 2024.1 though, unfortunately.

If you're able to preprocess the incoming data somehow, sending `@mt` instead of `@m`, and replacing `{` with `{{` and `}` with `}}` in the messages would work around it (though it sounds like that might not be an option for you).

Hope this helps,
Nick

**Reply by chuseman** [reply]
Thanks for the quick response Nick.

I think I can actually implement this workaround on my side. I can't easily change the log messages themselves all over in legacy code, but I can escape the rendered message before sending it.

Are `{` and `}` the only characters I need to worry about escaping?

I've noticed when sending only `@mt`, if I **Export -> Copy raw JSON**, it shows as if the message has both `@m` and `@mt` with the escaped values in both. If I actually send both `@m` (raw) and `@mt` (escaped), Copy raw JSON shows what I sent. So I assume that's what I should be doing? At least until v2024.1 ships?

Thanks,
Chris

**Reply by nblumhardt** [reply]
> send both `@m` (raw) and `@mt` (escaped)

Yes, that's the best idea 👍
